### PR TITLE
Fixing Clang warnings

### PR DESCRIPTION
**Pull request by philbucher**
and adding them to the CI such that the build stays clean in the future

Now the core and many apps compile clean, but there is still some work to do

Especially the `[-Woverloaded-virtual]` warning happens A LOT, but fixing this will be in a different PR

### CODE REVIEW

**Comment by philbucher** [comment]
ok now finally it works
Thanks @roigcarlo 

@loumalouomega I modified slightly your code to get rid of the warnings, are you ok with this?

File: applications/FluidDynamicsApplication/custom_processes/shock_detection_process.h [hunk 1]
```diff
@@ -137,9 +137,6 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) ShockDetectionProcess : public Proc
     ///@name Life Cycle
     ///@{
 
-    /// Default constructor.
-    ShockDetectionProcess() = default;
-
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_processes/shock_detection_process.h:142]: I saw this one some days ago. Thanks.

**Comment by philbucher** [comment]
@roigcarlo can we merge this?
If in the meantime some more warnings appear we will have a minor Problem :)

**Review by roigcarlo — APPROVED** [review]
Sure

**Pull request merged**